packages/amplify_datastore/example/lib/main.dart

```diff
@@ -131,24 +140,67 @@ class _MyAppState extends State<MyApp> {
     });
   }
 
+  savePost() async {
+    try {
+      Post post = Post(
+          title: 'New Post being saved', rating: 15, created: DateTime.now());
+      await Amplify.DataStore.save(post);
+      Post newPost = post.copyWith(id: post.id, title: 'Updated Title');
+      await Amplify.DataStore.save(newPost, when: Post.RATING.eq(10));
+    } catch (e) {
+      print(e);
+    }
+  }
+
+  deletePost() async {
+    try {
+      await Amplify.DataStore.delete(
+          // replace 'id' for testing as needed
+          Post(id: 'f6c4be2d-9b1b-496e-b3bf-78035f0e6191', title: 'Title'));
+    } catch (e) {
+      print(e);
```

Comment:
Instead of catching, can you bubble these all the way up, so that they crash the thread?